crates/uv/tests/it/sync.rs

```diff
@@ -12756,12 +12756,16 @@ fn sync_build_dependencies_respect_locked_versions() -> Result<()> {
          |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       Dependencies marked with `match-runtime = true` cannot include version specifiers
 
-    error: Failed to parse: `pyproject.toml`
-      Caused by: TOML parse error at line 11, column 9
-       |
-    11 | child = [{ requirement = "anyio>4", match-runtime = true }]
-       |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-    Dependencies marked with `match-runtime = true` cannot include version specifiers
+    Resolved [N] packages in [TIME]
+      × Failed to build `child @ file://[TEMP_DIR]/child`
+      ├─▶ The build backend returned an error
+      ╰─▶ Call to `build_backend.build_wheel` failed (exit status: 1)
+
+          [stderr]
+          `EXPECTED_ANYIO_VERSION` not set
+
+          hint: This usually indicates a problem with the package or the build environment.
+      help: `child` was included because `parent` (v0.1.0) depends on `child`
```

Comment:
I guess this is "right" but it's kind of weird (we only warn, not fail, when we fail to parse options from a `pyproject.toml`). Before, we were both warning and erroring because these settings were parsed twice, once as `pyproject.toml` options and once as project settings.

Reply:
Oh shoot sorry, I forgot about this.